src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { setLocationsTask, SiteOSStep, WebsiteOS } from 'vscode-azureappservice';
+import { AzureWizardPromptStep, IWizardOptions } from 'vscode-azureextensionui';
+import { ext } from '../../../extensionVariables';
+import { localize } from '../../../localize';
+import { IFunctionAppWizardContext } from '../IFunctionAppWizardContext';
+import { getStackPicks } from './getStackPicks';
+
+export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWizardContext> {
+    public async prompt(context: IFunctionAppWizardContext): Promise<void> {
+        const placeHolder: string = localize('selectRuntimeStack', 'Select a runtime stack.');
+        context.newSiteStack = (await ext.ui.showQuickPick(getStackPicks(context), { placeHolder })).data;
+
+        if (!context.newSiteStack.minorVersion.stackSettings.linuxRuntimeSettings) {
+            context.newSiteOS = WebsiteOS.windows;
+        } else if (!context.newSiteStack.minorVersion.stackSettings.windowsRuntimeSettings) {
+            context.newSiteOS = WebsiteOS.linux;
+        } else if (!context.advancedCreation) {
```

Comment:
Shouldn't you default to the `preferredOs` first and if it's advanced creation, then check the `minorVersion.stackSettings`?

Reply:
Realistically the order here won't make a difference. The point of the above checks is to see if that stack even supports both OS's - if it only supports one OS, we just use that OS. It should be impossible for the "preferredOS" to be an OS that the stack doesn't support